datafusion/src/physical_optimizer/pruning.rs

```diff
@@ -586,8 +587,45 @@ fn build_predicate_expression(
                 .min_column_expr()?
                 .lt_eq(expr_builder.scalar_expr().clone())
         }
+        Operator::Like => {
+            match &**right {
+                // If the literal is a 'starts_with'
+                Expr::Literal(ScalarValue::Utf8(Some(string)))
+                    if !string.starts_with('%') =>
+                {
+                    let scalar_expr =
+                        Expr::Literal(ScalarValue::Utf8(Some(string.replace('%', ""))));
+                    // Behaves like Eq
+                    let min_column_expr = expr_builder.min_column_expr()?;
+                    let max_column_expr = expr_builder.max_column_expr()?;
+                    min_column_expr
+                        .lt_eq(scalar_expr.clone())
+                        .and(scalar_expr.lt_eq(max_column_expr))
+                }
+                _ => unhandled,
+            }
+        }
+        Operator::NotLike => {
+            match &**right {
+                // If the literal is a 'starts_with'
+                Expr::Literal(ScalarValue::Utf8(Some(string)))
+                    if !string.starts_with('%') =>
```

Comment:
What about patterns like `pat1%pat2` ?

Reply:
We should also consider escaped percent characters in the pattern. Example: `LIKE '100\% %'`

Reply:
I think as we can evaluate the like expression anyway, it might be easier to support like / not like to the full extent instead of only "startswith".

Reply:
I only focused on expressions that don't start with `%`, under the assumption that they would be a `starts_with`. I don't think we can support anything other than a `starts_with` because we translate the queries to `min LtEq value && value LtEq max`.

Or how would `LIKE '100\% %'` be evaluated?

Reply:
you are right, that makes sense 👍